directives/annotated.py

```diff
@@ -87,11 +92,26 @@ def depart_annotated_node(self, node):
     parsed_html = self.body  # extract generated feedback line
     self.body = env.redirect # restore original output
 
-    self.body.append(postprocess(u''.join(parsed_html), node['name']))
+    postprocessed_html = postprocess_annotation_tags(u''.join(parsed_html), node['name'])
+    postprocessed_html = postprocess_inline_annotations(postprocessed_html, node['name'])
+    self.body.append(postprocessed_html)
 
     self.body.append("</div>\n")
 
-def postprocess(html, annotation_id):
+def postprocess_inline_annotations(html, annotated_section_id):
+    inline_anno_count = 0
+
+    def make_annotation_span(match):
+        nonlocal inline_anno_count
+        inline_anno_count += 1
+        annotation_text = match.group(1)
+        bit_to_insert = match.group(2)
+        insert_attrib = ' insert="' + bit_to_insert + '"' if bit_to_insert else ""
```

Comment:
Seems like there is an HTML attribute `insert` here. It is not a standard attribute, so it should be `data-insert`. Data attributes are intended for custom data storage.

Reply:
That was only about the HTML. Sphinx/docutils nodes can use any arbitrary attributes.

Reply:
Thanks. I committed a change that uses a better name for that attribute throughout.